R/summarize_weighted_production.R

```diff
@@ -80,9 +80,9 @@ summarize_unweighted_production <- function(data, ...) {
     distinct() %>%
     group_by(.data$sector_ald, .data$technology, .data$year, ...) %>%
     # FIXME: Confusing: `weighted_production` holds unweighted_production?
-    summarize(weighted_production = .data$production / sum(.data$production), .groups = "keep") %>%
-    ungroup(.data$technology) %>%
-    mutate(weighted_technology_share = .data$weighted_production) %>%
+    summarize(weighted_production = .data$production, .groups = "keep") %>%
+    ungroup(.data$technology, .data$tmsr, .data$smsp) %>%
```

Comment:
@maurolepore I reverted your changes as they did not have the desired effect. `weighted_production` should contain the actual `production` values (I agree the naming is annoying and confusing. it was chosen to ensure that the rest of the reformatting in the code downstream uses the correct column regardless of the value of `weight_production`). 

The solution I show here is simply ungrouping by `tmsr` and `smsp` (which are passed in as grouping arguments via `summary_groups`. This seems to solve the test issue, though I don't think it's the most elegant solution. I tried just removing `tmsr` and `smsp` from the argument `summary_groups`, but it seems they are in fact necessary for other parts of the calculation.